### PR TITLE
[SPARK-29483][BUILD] Bump Jackson to 2.10.0

### PR DESCRIPTION
**Pull request by Fokko**
### What changes were proposed in this pull request?

Release blog: https://medium.com/@cowtowncoder/jackson-2-10-features-cd880674d8a2

Fixes the following CVE's:
https://www.cvedetails.com/cve/CVE-2019-16942/
https://www.cvedetails.com/cve/CVE-2019-16943/

Looking back, there were 3 major goals for this minor release:

- Resolve the growing problem of “endless CVE patches”, a stream of fixes for reported CVEs related to “Polymorphic Deserialization” problem (described in “On Jackson CVEs… ”) that resulted in security tools forcing Jackson upgrades. 2.10 now includes “Safe Default Typing” that is hoped to resolve this problem.
- Evolve 2.x API towards 3.0, based on changes that were done in master, within limits of 2.x API backwards-compatibility requirements.
- Add JDK support for versions beyond Java 8: specifically add“module-info.class” for JDK9+, defining proper module definitions for Jackson components

Full changelog: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.10

Improved Scala 2.13 support: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.10#scala

### Why are the changes needed?

Patches CVE's reported by the vulnerability scanner.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Ran `mvn clean install -DskipTests` locally.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
Please polish the PR description, @Fokko .

Also, cc @srowen since this is `Scala 2.13`-related.

**Comment by dongjoon-hyun** [comment]
BTW, @Fokko . We use `[BUILD]` instead of `[CORE]` for this kind of PRs~

File: dev/deps/spark-deps-hadoop-3.2 [hunk 1]
```diff
@@ -102,17 +102,18 @@ httpclient-4.5.6.jar
...
 jackson-jaxrs-base-2.9.5.jar
 jackson-jaxrs-json-provider-2.9.5.jar
 jackson-mapper-asl-1.9.13.jar
-jackson-module-jaxb-annotations-2.9.10.jar
-jackson-module-paranamer-2.9.10.jar
-jackson-module-scala_2.12-2.9.10.jar
+jackson-module-jaxb-annotations-2.10.0.jar
+jackson-module-paranamer-2.10.0.jar
+jackson-module-scala_2.12-2.10.0.jar
+jakarta.activation-api-1.2.1.jar
```
Comment (srowen) [inline dev/deps/spark-deps-hadoop-3.2]: We'll need an entry in `LICENSE-binary` for this, under the Eclipse Dist License section. (We seem to already have a copy of the license though in `licenses-binary`).

**Review by srowen — CHANGES_REQUESTED** [review]

File: dev/deps/spark-deps-hadoop-3.2 (hunk 1 above)
Comment (Fokko) [reply dev/deps/spark-deps-hadoop-3.2]: Done, thanks for pointing out

**Review by srowen — APPROVED** [review]
Looks OK pending tests.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @Fokko and @srowen .
Merged to master.

**Comment by dongjoon-hyun** [comment]
Could you make a back porting PR, @Fokko ?

**Comment by dongjoon-hyun** [comment]
cc @jiangxb1987 since he is the release manager for `3.0.0-preview`.

**Comment by srowen** [comment]
Not a back port to 2.4 right? We can't update Jackson in 2.4. I don't think we have to back port to the preview branch, even if we keep it.

**Comment by dongjoon-hyun** [comment]
Oh, I see. Got it, @srowen .

**Comment by krishna-pandey** [comment]
@srowen What's the issue in upgrading Jackson in 2.4?

**Comment by Fokko** [comment]
@krishna-pandey please refer to https://github.com/apache/spark/pull/21596

Jackson is notorious for having to change their public API, and aggressively deprecating methods. This will conflict with older versions of Hadoop. This might have huge implications for your application.

**Pull request closed without merging**